### ISSUE TITLE
Upgrade com.auth0 java-jwt dependency

### ISSUE THREAD

**Issue by derickson82** [Nexmo/nexmo-java#247]
The `java-jwt` `v2.x` library from Auth0 is distributed as a shaded jar. One of the dependencies packaged with it is `jackson-databind` that is a very old version with known security vulnerabilities. There are `v3.x` releases that are using more recent versions of jackson.

I could see one or more of the following actions being taken to address this issue:

* Upgrade to the `v3.x` line (probably the easiest)
* Ask Auth0 to release a non-shaded version of their jar, to allow consumers more control over dependency management. (probably not easily done in the near term)
* Switch to an alternative JWT library that allows for better dependency management (a lot harder, I know, but I thought I would throw it out there)

**Comment by cr0wst** [comment]
I took a look at this for a bit today and I noticed that once they went to v3.x they dropped the shaded jar. Right now, our JWT implementation is rather limited and it's something that I have plans to improve on.  

Unfortunately, I can't update the dependency to `v3.x` without bumping the major version as per [Semver](https://semver.org/). However, I have a few comments on this being flagged as a vulnerability:

- I just ran Snyk and will be releasing a patch with updated dependencies to the highest compatible version.
- The usage of the vulnerable version of Jackson Databind is restricted to only what https://github.com/auth0/java-jwt is using due to the shading.
- I will be producing a major version in the next couple of weeks to add support for a few things that I can't do while maintaining backwards compatibility. I will try to re-configure our JWT usage and add some additional features for it possibly even removing the dependency on this library, or at least upgrading past the shaded version.

Hope this helps!
Steve
